src/modules/PublicLab.TitleModule.Related.js

```diff
@@ -35,28 +38,73 @@ module.exports = function relatedNodes(module) {
+    module.el.find('.ple-module-content').append('<div style="display:none;" class="ple-title-added"></div>');
+    addedRelatedEl = module.el.find('.ple-title-added');
 
   }
 
+  function showAdded(addedResult, relatedResults) {
+    addedRelatedEl.append('<div class="addedresult addedresult-' + addedResult.id + '" style="margin: 3px;"><a class="btn btn-xs btn-default remove-tag"><i class="fa fa-times-circle"></i> Remove</a> <a class="addedtitle"></a> by <a class="addedauthor"></a></div>');
+    addedRelatedEl.find('.addedresult-' + addedResult.id + ' .addedtitle').html(addedResult.title);
+    addedRelatedEl.find('.addedresult-' + addedResult.id + ' .addedtitle').attr('href', addedResult.url);
+    addedRelatedEl.find('.addedresult-' + addedResult.id + ' .addedauthor').html('@' + addedResult.author);
+    addedRelatedEl.find('.addedresult-' + addedResult.id + ' .addedauthor').attr('href', '/profile/' + addedResult.author);
+
+    $('.addedresult-' + addedResult.id + ' .remove-tag').click(function() {
+      var selectedToken = (editor.tagsModule.el.find('.token[data-value="response:' + addedResult.id +'"]'));
+      var y_axis = window.scrollY;
+      selectedToken.find('.close').trigger('click');
+      window.scrollTo(window.scrollX, y_axis);
+      addedRelatedPost = addedRelatedPost.replace(addedResult.id, '');
+      $('.addedresult-' + addedResult.id).remove();
+      show(relatedResults);
+      if(addedRelatedPost==='addedTitles') {
+        $('.related-post').remove();
+      }
+    });
+  }
+
   // expects array of results in format:
   // { id: 3, title: 'A third related post', url: '/', author: 'bsugar'}
   function show(relatedResults) { 
-
     relatedEl.find('.result').remove();
+    // If tag removed from TagModule
+    editor.tagsModule.el.find('input').on('tokenfield:removedtoken', function(e) {
+      var id = (e.attrs.value).replace('response:', '');
+      if(addedRelatedPost.includes(id)) {
+        addedRelatedPost = addedRelatedPost.replace(id, '');
+        $('.addedresult-' + id).remove();
+        show(relatedResults);
+        if(addedRelatedPost==='addedTitles') {
+          $('.related-post').remove();
+        }
+      }
+    });
 
     relatedResults.slice(0, 8).forEach(function(result) {
-
-      relatedEl.append('<div class="result result-' + result.id + '" style="margin: 3px;"><a class="btn btn-xs btn-default add-tag"><i class="fa fa-plus-circle"></i> Add</a> <a class="title"></a> by <a class="author"></a></div>');
-      relatedEl.find('.result-' + result.id + ' .title').html(result.title);
-      relatedEl.find('.result-' + result.id + ' .title').attr('href', result.url);
-      relatedEl.find('.result-' + result.id + ' .author').html('@' + result.author);
-      relatedEl.find('.result-' + result.id + ' .author').attr('href', '/profile/' + result.author);
+      var showRealted = false;
```

Comment:
`showRelated`

Reply:
@VladimirMikulic  I have applied the preetier format on the changes nothing has changed except some space after `function ()` but we are using `function()` in all the places.